core/trino-parser/src/main/java/io/trino/sql/SqlFormatter.java

```diff
@@ -354,19 +378,21 @@ protected Void visitPrepare(Prepare node, Integer indent)
         protected Void visitDeallocate(Deallocate node, Integer indent)
         {
             append(indent, "DEALLOCATE PREPARE ");
-            builder.append(node.getName());
+            builder.append(formatExpression(node.getName()));
```

Comment:
Does it make sense to test this and others?

Reply:
Idk. Does it make sense to add tests for these statements which happened to be buggy (assuming this was a bug), but not for these that happened not to be buggy? What tests would we write if we were writing this class from scratch today?

Reply:
We used to have the `SqlFormatter` tested via tests in `TestSqlParser`, which among others tested the parsing-formatting roundtrip. However, those testing methods are now marked deprecated in favor of other tests which do not test the roundtrip.

Reply:
> those testing methods are now marked deprecated in favor of other tests which do not test the roundtrip.

if so, why were they deprecated?

Reply:
From the commit message:
```
Mark assertExpression and assertStatement as deprecated  There's a much better and more flexible alternative with ParserAssert.
```
The preexisting tests were not rewritten to use ParserAssert.

Reply:
This looks like coming from b0638b509a0965af74731fc921a37ba849fa0334 https://github.com/trinodb/trino/pull/7519

but if the new method doesn't test roundtrip, allowing for bugs to sneak it, is it "much better"?
or maybe we can fix that?

Reply:
Ah, it _does_ validate SQL -> AST -> SQL roundtrip. My bad.